GWMiniGame/GWMiniGame.cpp

```diff
@@ -0,0 +1,287 @@
+			dates.push_back(d);
+		}
+
+		if (dates.empty()) {
+			file.close();
+			return true;
+		}
+		else {
+			for (int i = 0; i < dates.size(); i++) {
+				date = dates[i];
+				if (date == getDate()) {
+					file.close();
+					return false;
+				}
+			}
+			file.close();
+			return true;
+		}
+	}
+}
+
+int letterCount(const std::string& array, const char currentChar) {
+	int count = 0;
+	for (int i = 0; i < array.length(); ++i) {
+		char lowerCaseDB = std::tolower(array[i]);
+		if (lowerCaseDB == currentChar) {
+			count++;
+		}
+	}
+	return count;
+}
+
+bool findIndex(const std::string& array, const char* arrayInput, const char currentChar) {
+	bool result = false;
+	const int maxWordLength = 5;
+	const int falseValue = 9;
+	int indexDB[maxWordLength];
+	int indexInput[maxWordLength];
+	for (int i = 0; i < maxWordLength; ++i) {
+		if (array[i] == currentChar) {
+			indexDB[i] = i;
+		}
+		else {
+			indexDB[i] = falseValue;
+		}
+	}
+	for (int i = 0; i < maxWordLength; ++i) {
+		if (arrayInput[i] == currentChar) {
+			indexInput[i] = i;
+		}
+		else {
+			indexInput[i] = falseValue;
+		}
+	}
+	for (int i = 0; i < maxWordLength; ++i) {
+		if (indexDB[i] != falseValue) {
+			for (int j = 0; j < maxWordLength; ++j) {
+				if (indexDB[i] == indexInput[j]) {
+					result = true;
+					return result;
+				}
+			}
+		}
+	}
+	return result;
+}
+
+bool isWordGuessed(const std::string& arrayDB, const char* arrayInput) {
+	for (int i = 0; i < arrayDB.length() && arrayInput[i] != '\0'; ++i) {
+		char lowerCaseDB = std::tolower(arrayDB[i]);
+		char lowerCaseInput = std::tolower(arrayInput[i]);
+		if (lowerCaseDB != lowerCaseInput) {
+			return false;
+		}
+	}
+	return true;
+}
+
+char* wordCompare(const std::string& arrayDB, const char* arrayInput) {
+	const int maxWordLength = 5;
+	char countCheck[maxWordLength + 1];
+	char* result = new char[maxWordLength + 1];
+	
+	for (int i = 0; i < arrayDB.length(); ++i) {
+		char lowerCaseDB = std::tolower(arrayDB[i]);
+		char lowerCaseInput = std::tolower(arrayInput[i]);
+
+		if (lowerCaseDB == lowerCaseInput) {
+			countCheck[i] = lowerCaseInput;
+			result[i] = (char)std::toupper(lowerCaseInput);
+		}
+		else {
+			countCheck[i] = lowerCaseInput;
+			if (letterCount(countCheck, lowerCaseInput) <= letterCount(arrayDB, lowerCaseInput) && letterCount(arrayDB, lowerCaseInput) == 1) {
+				if (findIndex(arrayDB, arrayInput, lowerCaseInput)) {
+					result[i] = '*';
+				}
+				else {
+					result[i] = lowerCaseInput;
+				}
+			}
+			else if(letterCount(countCheck, lowerCaseInput)<= letterCount(arrayDB, lowerCaseInput) && letterCount(arrayDB, lowerCaseInput)>1){
+				result[i] = lowerCaseInput;
+			}
+			else {
+				result[i] = '*';
+			}
+		}
+	}
+	result[5] = '\0';
+	return result;
+}
+
+int main(){
+	while (true) {
+		int value = mainMenu();
+		if (value == 0) {
+			std::cout << "See You next time!" << std::endl;
+			break;
+		}
+		if (value == 1) {
+		const std::string& randomWord = randomWordFromBD("WordDB.txt");
```

Comment:
since you already use std::string here it's ok to use it everywhere instead of char* . It's not all-round better but in your cases there's no noticable benefits of using raw char* strings.